extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/SmallRyeReactiveMessagingKafkaProcessor.java

```diff
@@ -26,4 +40,492 @@ public void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        return DotNames.PUBLISHER_BUILDER.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 1;
+    }
+
+    private static boolean isProcessor(Type type) {
+        // raw type Processor is wrong, must be Processor<Something, SomethingElse>
+        return DotNames.PROCESSOR.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 2;
+    }
+
+    private static boolean isProcessorBuilder(Type type) {
+        // raw type ProcessorBuilder is wrong, must be ProcessorBuilder<Something, SomethingElse>
+        return DotNames.PROCESSOR_BUILDER.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 2;
+    }
+
+    // ---
+
+    private static boolean isEmitter(Type type) {
+        // raw type Emitter is wrong, must be Emitter<Something>
+        return DotNames.EMITTER.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 1;
+    }
+
+    private static boolean isMutinyEmitter(Type type) {
+        // raw type MutinyEmitter is wrong, must be MutinyEmitter<Something>
+        return DotNames.MUTINY_EMITTER.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 1;
+    }
+
+    // ---
+
+    private static boolean isMessage(Type type) {
+        // raw type Message is wrong, must be Message<Something>
+        return DotNames.MESSAGE.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 1;
+    }
+
+    private static boolean isKafkaRecord(Type type) {
+        // raw type KafkaRecord is wrong, must be KafkaRecord<Something, SomethingElse>
+        return DotNames.KAFKA_RECORD.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 2;
+    }
+
+    private static boolean isRecord(Type type) {
+        // raw type Record is wrong, must be Record<Something, SomethingElse>
+        return DotNames.RECORD.equals(type.name())
+                && type.kind() == Type.Kind.PARAMETERIZED_TYPE
+                && type.asParameterizedType().arguments().size() == 2;
+    }
+
+    private static boolean isRawMessage(Type type) {
+        switch (type.kind()) {
+            case PRIMITIVE:
+            case CLASS:
+            case ARRAY:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    // ---
+
+    // TODO can we use Map.ofEntries(...) now that we're on 11?
```

Comment:
I'd really like an answer here :-)

Reply:
Yes

Reply:
Gonna change that then, thanks!

Reply:
As long as you don't plan on getting a patch backported, you can use Java 11 goodness.

Reply:
I don't think we do. Do we, @cescoffier?